### PR TITLE
Nigao/readselection

### PR DESCRIPTION
**Pull request by ninggao**
(1) add new feature to read back the value of a property, then ask if a change/add is needed.
(2) add unexpected set message to all (string, number, object) set.
(3) remove keyPhrase, propertyName, personName from triggerIntent to enable other intents.

### CODE REVIEW

File: packages/dialog/templates/numberSetdimension.dialog.lg [hunk 1]
```diff
@@ -10,11 +10,18 @@
...
     "entity": "dimension",
     "actions": [
+        {
+            "$kind": "Microsoft.SendActivity",
+            "activity": "@{callSetMessage()}"
+        },
         {
             "$kind": "Microsoft.SetProperty",
             "property": "$@{property}",
             "value": "@dimension"
```
Comment (chrimc62) [inline packages/dialog/templates/numberSetdimension.dialog.lg]: >@dimension [](start = 22, length = 10)

I think this needs to be dimension.number, right?

File: packages/dialog/templates/stringSettemperature.dialog.lg [hunk 2]
```diff
@@ -10,11 +10,18 @@
...
     "entity": "temperature",
     "actions": [
+        {
+            "$kind": "Microsoft.SendActivity",
+            "activity": "@{callSetMessage()}"
+        }, 
         {
             "$kind": "Microsoft.SetProperty",
             "property": "$@{property}",
             "value": "@temperature"
```
Comment (chrimc62) [inline packages/dialog/templates/stringSettemperature.dialog.lg]: >@temperature [](start = 22, length = 12)

Is this right?

File: packages/dialog/templates/en-us/readPropertyIntent.lu [hunk 3]
```diff
@@ -0,0 +1,8 @@
+# ReadProperty
+- what's my current selection of {PROPERTYName}
+- what's the curernt value of {PROPERTYName}
```
Comment (chrimc62) [inline packages/dialog/templates/en-us/readPropertyIntent.lu]: >curernt [](start = 13, length = 7)

spelling

File: packages/dialog/templates/en-us/readPropertyIntent.lu [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+- what's the value of {PROPERTYName}
+- value of {PROPERTYName}
+- the value of {PROPERTYName}
+- selection of {PROPERTYName}
+- current selection of {PROPERTYName}
```
Comment (chrimc62) [inline packages/dialog/templates/en-us/readPropertyIntent.lu]: >of [](start = 20, length = 2)

Maybe [of | for] for all of these.

File: packages/dialog/templates/ChangePropertyConfirmationSet.dialog.lg [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        {
+            "$kind": "Microsoft.IfCondition",
+            "condition":"$ChangePropertyConfirmation == 'yes'",
+            "actions": [
+                {
+                    "$kind": "Microsoft.Ask",
+                    "activity": "@{callAsk()}",
+                    "expectedProperties": [
+                        "$PropertyToRemember"
+                    ]
```
Comment (chrimc62) [inline packages/dialog/templates/ChangePropertyConfirmationSet.dialog.lg]: Should be "createArray($PropertyToRemember)".

File: packages/dialog/templates/en-us/triggerIntent.lu.lg [hunk 6]
```diff
@@ -7,6 +7,12 @@
 # entityPattern(entity)
 - IF: @{equals(entity, 'utterance')}
 - 
+- ELSEIF: @{equals(entity, 'personName')}
+- 
+- ELSEIF: @{equals(entity, 'keyPhrase')}
+- 
+- ELSEIF: @{equals(entity, 'PROPERTYName')}
+-
```
Comment (chrimc62) [inline packages/dialog/templates/en-us/triggerIntent.lu.lg]: I think you should be able to check if it is in the schema.$expectedOnly and if so exclude it.

File: packages/dialog/templates/numberSetdimension.dialog.lg (hunk 1 above)
Comment (ninggao) [reply packages/dialog/templates/numberSetdimension.dialog.lg]: File deleted.

File: packages/dialog/templates/stringSettemperature.dialog.lg (hunk 2 above)
Comment (ninggao) [reply packages/dialog/templates/stringSettemperature.dialog.lg]: File deleted.

File: packages/dialog/templates/en-us/readPropertyIntent.lu (hunk 3 above)
Comment (ninggao) [reply packages/dialog/templates/en-us/readPropertyIntent.lu]: Revised.

File: packages/dialog/templates/en-us/readPropertyIntent.lu (hunk 4 above)
Comment (ninggao) [reply packages/dialog/templates/en-us/readPropertyIntent.lu]: Added.

File: packages/dialog/templates/ChangePropertyConfirmationSet.dialog.lg (hunk 5 above)
Comment (ninggao) [reply packages/dialog/templates/ChangePropertyConfirmationSet.dialog.lg]: Done.

File: packages/dialog/templates/en-us/triggerIntent.lu.lg (hunk 6 above)
Comment (ninggao) [reply packages/dialog/templates/en-us/triggerIntent.lu.lg]: Done.

File: packages/dialog/templates/en-us/readPropertyIntent.lu [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+- what's the current value of {PROPERTYName}
+- what's the value of {PROPERTYName}
+- value of {PROPERTYName}
+- the value of {PROPERTYName}
+- selection of {PROPERTYName}
+- current selection of {PROPERTYName}
+- value for {PROPERTYName}
+- the value for {PROPERTYName}
+- selection for {PROPERTYName}
+- current selection for {PROPERTYName}
```
Comment (chrimc62) [inline packages/dialog/templates/en-us/readPropertyIntent.lu]: You could have done this by "the value (of | for) {PROPERTYName}"

File: packages/dialog/templates/en-us/common.lg [hunk 8]
```diff
@@ -59,7 +59,7 @@ Enter a value for @{name(property)}
 
 # requiredProperties
 - IF: @{count(dialogClass.schema.required) > 0}
-- Required properties include: @{join(foreach(where(dialogClass.schema.required, property, !contains(dialogClass.schema.$expectedOnly, property)), property, name(property)), ',')}\n
+- Required properties include: @{join(foreach(dialogClass.schema.required, property, name(property)), ',')}\n
```
Comment (chrimc62) [inline packages/dialog/templates/en-us/common.lg]: >dialogClass.schema.required [](start = 46, length = 27)

This is as good as we can do, but will need to think about if we can do better.

**Pull request merged**